### PR TITLE
Fix texcoords in terrain shadowpass when _ALPHATEST_ON is set

### PR DESCRIPTION
**Pull request by nerites**
### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [n/a] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [x] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Fix for https://fogbugz.unity3d.com/f/cases/1349305/
Since _ALPHATEST_ON was converted from a multi_compile to a multi_compile_fragment, it's never set during the vertex shader. Because of this, the texcoords used to sample the holes texture weren't getting passed to the fragment shader properly in the shadow pass. This PR removes _ALPHATEST_ON ifdefs from struct declarations and vertex shaders, under the assumption that it's more important to reduce shader variants than to optimize vertex shader performance.

ShadowPassVertex also must pass UVs into TerrainInstancing now since the UVs must be correctly updated for the fragment shadowpass (this probably did not work before). 

---
### Testing status
Tested using the method described in the fogbugz, for both instanced and non-instanced terrain:
Before:
holesbug

After:
holesfixed

Modified Terrain Holes test to cover hole shadowing:
image

Rerunning tests to generate ref images:
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/terrain%252Fholeshadowfix/.yamato%252Fall-urp.yml%2523PR_URP_2021.2
---
### Comments to reviewers
Added Landon since he filed the original bug. Added Elvar since he was the one that added _ALPHATEST ifdefs into the vert shaders.

### CODE REVIEW

**Comment by LandonTownsendUnity** [comment]
Should have a graphics test to prevent this from regressing in the future.

**Comment by nerites** [comment]
Is a package out for 21.2 yet? Do I need to backport for that as well?

**Review by LandonTownsendUnity — APPROVED** [review]
Functionally works well, so I'll approve now that there's an automated test.

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - The behavior of setting a camera's Background Type to "Dont Care" has changed on mobile. Previously, "Dont Care" would behave identically to "Solid Color" on mobile. Now, "Dont Care" corresponds to the render target being filled with arbitrary data at the beginning of the frame, which may be faster in some situations. Note that there are no guarantees for the exact content of the render target, so projects should use "Dont care" only if they are guaranteed to render to, or otherwise write every pixel every frame.
 - Stripping shader variants per renderer features instead of combined renderer features.
 - When MSAA is enabled and a depth texture is required, the opaque pass depth will be copied instead of scheduling a depth prepass.
+- Added a light to the Terrain Holes test to cover hole shadowing
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/CHANGELOG.md:73]: Testing related changes are not necessary to be included in changelog as they are not user facing.

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/CHANGELOG.md (hunk 1 above)
Comment (nerites) [reply com.unity.render-pipelines.universal/CHANGELOG.md]: thanks, removed in 2f2606f!

**Comment by nerites** [comment]
All remaining test failures (URP_2D is failing and all iPhone tests are failing) are test failures that occur on latest master as well: 
https://unity-ci.cds.internal.unity3d.com/job/7754733

**Pull request merged**